juju/machine.py

```diff
@@ -239,6 +239,8 @@ def dns_name(self):
             scope = address['scope']
             if scope == 'public' or scope == 'local-cloud':
```

Comment:
Python's logic ops like `or` are short-circuiting (e.g. `1 or 1/0 or 3` will produce 1), so it might make it cleaner if you just add `or scope == 'local-fan'` at the end of this line.